torch/csrc/jit/ir/irparser.cpp

```diff
@@ -226,12 +229,41 @@ ParsedLiteral IRParser::parseScalarLiteral(Node* n) {
       r.k = AttributeKind::t;
       return r;
     }
+    case '{': {
+      L.next();
+      if (L.cur().kind == '-') {
+        L.next();
+      }
+      auto text = L.expect(TK_NUMBER);
+      if (!parse_tensor_constants_) {
+        throw ErrorReport(token.range)
+            << "Single-element tensor constant encoutered but "
+            << "`parse_tensor_constants` is set to false " << token.text();
+      }
+      L.expect('}');
+      deferred_tensor_value_initializations_.push_back(n);
+      r.k = AttributeKind::t;
+      return r;
+    }
     default:
       throw ErrorReport(token.range)
           << "Could not parse literal" << token.text();
   }
 }
 
+void IRParser::bypassTypeAnnotationList() {
+  int depth = 0;
+  while (!(L.cur().kind == ']' && depth == 1)) {
```

Comment:
nit: felt the condition here took a little bit to parse. would it be simpler to do something like 
```
bool bypassed_list = false;
int depth = 0;
while (depth != 0 &&  !bypassed_list) {
    if (L.cur().kind == '[') {
      bypassed_list = True;
      depth++;
    } else if (L.cur().kind == ']') {
      depth--;
    }
    L.next();
  }
}
```